### PR TITLE
Progress bar new design

### PR DESCRIPTION
**Pull request by Inbal-Tish**
## Description
ProgressBar - new component design and API.

## Changelog
ProgressBar - new component design and API.

### CODE REVIEW

File: src/components/progressBar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+      ...progressStyle
+    };
+  }
+
+  renderCustomElement() {
+    const {customElement} = this.props;
+
+    if (customElement) {
+      return React.cloneElement(customElement, {
+        style: [customElement.props.style, styles.progress]
```
Comment (ethanshar) [inline src/components/progressBar/index.tsx:133]: I don't customElement has `props.style` 
Are you sure you're doing this correctly?

Comment (Inbal-Tish) [reply src/components/progressBar/index.tsx:133]: What do you mean? Use optional chaining?

Comment (ethanshar) [reply src/components/progressBar/index.tsx:133]: Not exactly.. 
Does `customElement` has `props` on it.. after all we're passing a JSX, no?

Comment (Inbal-Tish) [reply src/components/progressBar/index.tsx:133]: Well, yes. JSX extends ReactElement, which has 'props'

Comment (Inbal-Tish) [reply src/components/progressBar/index.tsx:133]: What is the use case you're thinking about?

**Review by ethanshar — APPROVED** [review]

**Pull request merged**